releases/net471/KnownIssues/534719-Networking ServicePoint.ConnectionLimit default behavior with loopback changed unexpectedly.md

```diff
@@ -0,0 +1,73 @@
+non-loopback addresses such as http://www.microsoft.com.  For loopback addresses such as http://localhost, the limit
+for connections was always int.MaxValue (2,147,483,647) unless changed by calling `ServicePoint` class specific APIs.
+
+Due to this bug, when using HttpClient APIs, the connection limit for loopback addresses will now match the limit for
+non-loopback addresses. Thus, the default limit would be 2. This can cause applications to run slower or hang when
+doing multiple, parallel, requests to the http://localhost addresses.
+
+Here is code that can show the problem:
+
+```c#
+using System;
+using System.Net;
+using System.Net.Http;
+
+namespace ConsoleApp
+{
+    class Program
+    {
+        static void Main(string[] args)
+        {
+            Console.WriteLine($"ServicePointManager.DefaultConnectionLimit: {ServicePointManager.DefaultConnectionLimit}");
+
+            var uriLoopback = new Uri("http://localhost");
+            var uriExternal = new Uri("http://www.microsoft.com");
+
+            Console.WriteLine("Before using HttpClient APIs");
+            ServicePoint spLoopback = ServicePointManager.FindServicePoint(uriLoopback);
+            Console.WriteLine($"{uriLoopback.AbsoluteUri}, ConnectionLimit (should be {int.MaxValue}): {spLoopback.ConnectionLimit}");
+
+            ServicePoint spExternal = ServicePointManager.FindServicePoint(uriExternal);
+            Console.WriteLine($"{uriExternal.AbsoluteUri}, ConnectionLimit (should be {ServicePointManager.DefaultConnectionLimit}): {spExternal.ConnectionLimit}");
+
+            Console.WriteLine("Use HttpClient APIs");
+            var client = new HttpClient();
+            try
+            {
+                HttpResponseMessage response = client.GetAsync(uriLoopback).Result;
+            }
+            catch (Exception)
+            {
+                // Ignore any network error since there is probably not a loopback server present.
+            }
+
+            Console.WriteLine("After using HttpClient APIs");
+
+            // BUG - due to the bug in .NET Framework 4.7.1, the ConnectionLimit for this loopback ServicePoint is changed
+            // unexpectedly.
+            Console.WriteLine($"{uriLoopback.AbsoluteUri}, ConnectionLimit (should be {int.MaxValue}): {spLoopback.ConnectionLimit}");
+        }
+    }
+}
+```
+
+## Cause
+Changes in .NET Framework 4.7.1 to the `System.Net.Http.HttpClientHandler` class caused this problem.
+
+## Resolution
+To work around the problem using HttpClient APIs, you can use the following code to increase the connection limit.
+This code should be added before calling any HttpClient APIs.
+
+```c#
+ServicePointManager.DefaultConnectionLimit = 20; // Actual value should be based on your requirements.
+```
+
+Note: This will also change the limits for non-loopback addresses as well.
+
+## More information
+This problem is planned to be fixed with a future .NET Framework Quality Rollup release.
```

Comment:
This problem is planned to be fixed --> very tentative
Alternatives:
We plan to fix this problem in...
We will fix this problem in.... (the best alternative, if it in fact will be fixed)

Reply:
Fixed.